Asterix/Optical_System_functions.py

```diff
@@ -829,10 +878,10 @@ def EF_through(self,
                     int(wavelength * 1e9))
                 useful.save_plane_in_fits(dir_save_all_planes, name_plane,
                                           corono_focal_plane)
-
-                name_plane = 'FPM' + '_wl{}'.format(int(wavelength * 1e9))
-                useful.save_plane_in_fits(dir_save_all_planes, name_plane,
-                                          FPmsk)
+                if not noFPM:
+                    name_plane = 'FPM' + '_wl{}'.format(int(wavelength * 1e9))
+                    useful.save_plane_in_fits(dir_save_all_planes, name_plane,
+                                              FPmsk)
 
                 name_plane = 'EF_FP_after_FPM' + '_wl{}'.format(
```

Comment:
name_plane écrase celui défini dans "if not noFPM" deux lignes plus haut non ?

Reply:
yes but i saved the thing I want in between